src/integrate/orchestrator.rs

```diff
@@ -0,0 +1,1204 @@
+        port_bindings.insert(
+            format!("{}/tcp", devnet_config.bitcoin_explorer_port),
+            Some(vec![PortBinding {
+                host_ip: Some(String::from("0.0.0.0")),
+                host_port: Some(format!("{}/tcp", devnet_config.bitcoin_explorer_port)),
+            }]),
+        );
+
+        let mut exposed_ports = HashMap::new();
+        exposed_ports.insert(format!("{}/tcp", devnet_config.bitcoin_explorer_port), HashMap::new());
+
+        let mut labels = HashMap::new();
+        labels.insert("project".to_string(), self.network_name.to_string()); 
+
+        let config = Config {
+            labels: Some(labels),
+            image: Some(devnet_config.bitcoin_explorer_image_url.clone()),
+            domainname: Some(self.network_name.to_string()),
+            tty: Some(true),
+            exposed_ports: Some(exposed_ports),
+            env: Some(vec![
+                format!(
+                    "BTCEXP_HOST=0.0.0.0",
+                ),
+                format!(
+                    "BTCEXP_PORT={}", devnet_config.bitcoin_explorer_port
+                ),
+                format!(
+                    "BTCEXP_BITCOIND_HOST=host.docker.internal",
+                ),
+                format!(
+                    "BTCEXP_BITCOIND_PORT={}", devnet_config.bitcoin_node_rpc_port
+                ),
+                format!(
+                    "BTCEXP_BITCOIND_USER={}", devnet_config.bitcoin_node_username
+                ),
+                format!(
+                    "BTCEXP_BITCOIND_PASS={}", devnet_config.bitcoin_node_password
+                ),
+            ]),
+            host_config: Some(HostConfig {
+                port_bindings: Some(port_bindings),
+                ..Default::default()
+            }),
+            ..Default::default()
+        };
+
+        let options = CreateContainerOptions {
+            name: format!("bitcoin-explorer.{}", self.network_name),
+        };
+
+        let container = docker
+            .create_container::<String, String>(Some(options), config)
+            .await
+            .map_err(|e| format!("Unable to create container: {}", e))?
+            .id;
+
+        self.bitcoin_explorer_container_id = Some(container.clone());
+
+        docker
+            .start_container::<String>(&container, None)
+            .await
+            .map_err(|e| format!("Unable to create container: {}", e))?;
+
+        let res = docker
+            .connect_network(
+                &self.network_name,
+                ConnectNetworkOptions {
+                    container,
+                    ..Default::default()
+                },
+            )
+            .await;
+
+        if let Err(e) = res {
+            let err = format!("Error connecting container: {}", e);
+            println!("{}", err);
+            return Err(err);
+        }
+
+        Ok(())
+    }
+
+    pub async fn restart(&mut self) {}
+
+    pub async fn terminate(&mut self) {
```

Comment:
It might be useful to incorporate a `stop` function to allow devs to stop their devnet containers without terminating them and losing progress on their local cluster. This is possible if the dev uses the `docker` CLI directly, but would be a nice-to-have if devs can do this while in `devnet` land :)